src/com/apeshko/javacore/task_01/Application.java

```diff
@@ -0,0 +1,244 @@
+
+        flyingMachines
+                .stream()
+                .forEach(flyingMachine -> {
+                    System.out.println(" - " + flyingMachine.getName() + " " + flyingMachine.getDistance());
+                });
+    }
+
+    public static void searchByCriteriaTest() {
+        final AirCompanySearchService airCompanySearchService = new AirCompanySearchServiceImpl();
+        final AirCompany airCompany = getRandomAirCompany();
+        final FlyingMachineSearchCriteria flyingMachineSearchCriteria = new FlyingMachineSearchCriteria.Builder()
+                .withCapacityStart(50)
+                .withCapacityEnd(120)
+                .build();
+
+        List<FlyingMachine> flyingMachines = airCompanySearchService.search(airCompany, flyingMachineSearchCriteria);
+
+        System.out.println("Flying machines of the company " + airCompany.getName() + " with capacity in a range [" + flyingMachineSearchCriteria.getCapacityStart() + ", " + flyingMachineSearchCriteria.getCapacityEnd() + "]:");
+        flyingMachines
+                .stream()
+                .forEach(flyingMachine -> {
+                    System.out.println(" - " + flyingMachine.getName() + " " + flyingMachine.getCapacity());
+                });
+    }
+
+    static void makeFlightSuccessTest() {
+        final FlightServiceImpl flightService = new FlightServiceImpl();
+
+        final Airport airportFrom = new Airport("Minsk", 1, 1);
+        final Airport airportTo = new Airport("Antuan de Sent Exupery", 1000, 1000);
+        final FlyingMachine airplane = new Airplane.Builder()
+                .withName("Sea")
+                .withCapacity(120)
+                .withType(Airplane.Type.AIRBUS)
+                .withHeight(1200)
+                .withWidth(350)
+                .withLength(24567.9)
+                .withWeight(3400)
+                .withDistance(6000)
+                .build();
+
+        try {
+            flightService.makeFlight(airportFrom, airportTo, airplane);
+        } catch (NotEnoughDistanceException e) {
+            System.err.println(e);
+        }
+    }
+
+    static void makeFlightExceptionTest() {
+        final FlightServiceImpl flightService = new FlightServiceImpl();
+
+        final Airport airportFrom = new Airport("Minsk", 1, 1);
+        final Airport airportTo = new Airport("Antuan de Sent Exupery", 1000, 1000);
+        final FlyingMachine airplane = new Quadcopter.Builder()
+                .withName("Mass Attack")
+                .withCapacity(0)
+                .withType(Quadcopter.Type.MILITARY)
+                .withHeight(100)
+                .withWidth(100)
+                .withLength(100.9)
+                .withWeight(20)
+                .withDistance(500)
+                .build();
+
+        try {
+            flightService.makeFlight(airportFrom, airportTo, airplane);
+        } catch (NotEnoughDistanceException e) {
+            System.err.println(e);
+        }
+    }
```

Comment:
try to avoid static, better to move all this methods to service class, inject this service, and call methods by object reference
typical use case for static: factory methods, builders, constants